azure-functions-java-sdktypes/src/main/java/com/microsoft/azure/functions/sdktype/blob/BlobContainerHydrator.java

```diff
@@ -0,0 +1,136 @@
+            LOGGER.info("Detected connection string usage from environment variable: " + envVar);
+            return buildWithConnectionString(containerName, maybeConnString);
+        } else {
+            // interpret envVar as prefix for managed identity
+            LOGGER.info("Detected managed identity usage. Prefix: " + envVar);
+
+            String accountName = System.getenv(envVar + "__accountName");
+            String serviceUri = System.getenv(envVar + "__serviceUri");
+            String blobServiceUri = System.getenv(envVar + "__blobServiceUri");
+            String clientId = System.getenv(envVar + "__clientId");
+
+            String endpoint = resolveEndpoint(accountName, serviceUri, blobServiceUri);
+            Object credential = buildManagedIdentityCredential(clientId);
+
+            return buildWithManagedIdentity(containerName, endpoint, credential);
+        }
+    }
+
+    private boolean isConnectionString(String val) {
+        return val.contains("AccountKey=") || val.contains("DefaultEndpointsProtocol=");
+    }
+
+    private Object buildWithConnectionString(String containerName, String connStr) throws Exception {
+        ClassLoader cl = Thread.currentThread().getContextClassLoader();
+        Class<?> builderClass = cl.loadClass("com.azure.storage.blob.BlobContainerClientBuilder");
+        Object builder = builderClass.getDeclaredConstructor().newInstance();
+
+        Method connMethod = builderClass.getMethod("connectionString", String.class);
+        connMethod.invoke(builder, connStr);
+
+        Method contMethod = builderClass.getMethod("containerName", String.class);
+        contMethod.invoke(builder, containerName);
+
+        Method buildM = builderClass.getMethod("buildClient");
+        Object containerClient = buildM.invoke(builder);
+        LOGGER.info("Successfully built BlobContainerClient using connection string approach.");
+        return containerClient;
+    }
+
+    private Object buildWithManagedIdentity(String containerName, String endpoint, Object credential) throws Exception {
+        LOGGER.info("buildWithManagedIdentity for container: " + containerName + " endpoint: " + endpoint);
+
+        ClassLoader cl = Thread.currentThread().getContextClassLoader();
+        Class<?> builderClass = cl.loadClass("com.azure.storage.blob.BlobContainerClientBuilder");
+        Object builder = builderClass.getDeclaredConstructor().newInstance();
+
+        // reflect builder.credential(tokenCredential)
+        final String unshadedName = "com.azure.core.credential.TokenCredential";
+        final String fallBackName = "com.microsoft.azure.functions.shaded.com.azure.core.credential.TokenCredential";
+        Class<?> tokenCredClass = tryLoadClass(cl, unshadedName, fallBackName);
+        Method credMethod = builderClass.getMethod("credential", tokenCredClass);
+        credMethod.invoke(builder, credential);
+
+        Method endpointMethod = builderClass.getMethod("endpoint", String.class);
```

Comment:
Suggestion for 75-85

`
invokeMethodSafely(Object instance, Class<?> clazz, String methodName, Object... args) throws Exception {
        for (Method method : clazz.getMethods()) {
            if (method.getName().equals(methodName) && method.getParameterCount() == args.length) {
                return method.invoke(instance, args);
            }
        }
        throw new NoSuchMethodException("Method not found: " + methodName + " in class " + clazz.getName());
    }
    `